src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -347,6 +367,30 @@ public static class TagHelperBlockRewriter
             // Have a name now. Able to determine correct isBoundNonStringAttribute value.
             var result = CreateTryParseResult(name, descriptors);
 
+            var firstChild = builder.Children[0] as Span;
+            if (firstChild != null && firstChild.Symbols[0] is HtmlSymbol)
+            {
+                var htmlSymbol = firstChild.Symbols[firstChild.Symbols.Count - 1] as HtmlSymbol;
+                switch (htmlSymbol.Type)
+                {
+                    // Treat NoQuotes and DoubleQuotes equivalently. We purposefully do not persist NoQuotes
+                    // ValueStyles at code generation time to protect users from rendering dynamic content with spaces
+                    // that can break attributes.
+                    // Ex: <tag my-attribute=@value /> where @value results in the test "hello world".
```

Comment:
Sorry for the one-commit comment.

My earlier comment was that `<tag my-attribute=@value />` example can't be applicable both here and at [line 170](https://github.com/aspnet/Razor/pull/743/files#diff-72fec5f339f6e3ee604fc27d0fc6fd95R170). So what example applies here? Or, if this is for consistency with the earlier code, say that.